raft/src/main/java/org/apache/kafka/snapshot/RecordsSnapshotWriter.java

```diff
@@ -133,6 +133,28 @@ public static <T> Optional<SnapshotWriter<T>> createWithHeader(
         });
     }
 
+    public static <T> SnapshotWriter<T> createWithHeader(
```

Comment:
I think we can remove some code duplication if the function `public static <T> Optional<SnapshotWriter<T>> createWithHeader(...)` calls this implementation.